Profile audience and scope (#86)
smart-on-fhir/fhir-bulk-data-docs: authorization.md

# SMART Backend Services: Authorization Guide

  
## Profile Audience and Scope

This profile is intended to be used by developers of back-end services that
autonomously (or semi-autonomously) need to access FHIR resources from an EHR 
that has pre-authorized a defined scope of access.  Specifically, this profile 

> describes the registration process by which a backend service is pre-authorized, 

Reviewer: It does not really describe the registration process, does it?

Reviewer 2: It describes functional requirements for the process; would you prefer different wording here?

Reviewer: I prefer the old wording `This profile assumes that a backend service
has been authorized up-front`

Reviewer 2: OK. @bakerdb perhaps we can use a hybrid like:

```suggestion
describes the registration-time metadata required for a backend service to be pre-authorized,
```

Author: I don't think we provide much detail regarding the registration process at all.   We just require that the backend client has pre-registered and that the EHR has authorized a scope of access.

Author: I like your suggested wording here, Josh.

Reviewer 2: Cool  -- I think you can click to accept the proposed change (this is a new feature in GitHub; if you don't see it, just let me know and I'll apply directly).

Author: I see the "Apply Suggestion" button.  But when I click it, it says "Outdated Suggestions cannot be applied."  Guess I took too long.

Reviewer 2: Oh, ok -- you successfully accepted the suggestion yesterday, and it's now reflected in this PR. So... good :-)
